Sort with multiple syntatic list productions throws `AssertionError`

K version: `v6.3.26`

```k
// user-list.k
module USER-LIST
    imports BOOL-SYNTAX
    imports INT-SYNTAX

    syntax Foo ::= List{Bool, ","}
    syntax Foo ::= List{Int, ";"}
endmodule
```

```
$ kompile user-list.k --syntax-module USER-LIST --debug
java.lang.AssertionError
        at org.kframework.definition.UserList.getLists(UserList.java:49)
        at org.kframework.parser.inner.RuleGrammarGenerator.getCombinedGrammarImpl(RuleGrammarGenerator.java:735)
        at org.kframework.parser.inner.ParseInModule.getParsingModule(ParseInModule.java:183)
        at org.kframework.parser.inner.kernel.Scanner.<init>(Scanner.java:105)
        at org.kframework.parser.inner.ParseInModule.getScanner(ParseInModule.java:314)
        at org.kframework.kompile.DefinitionParsing.resolveNonConfigBubbles(DefinitionParsing.java:560)
        at org.kframework.kompile.DefinitionParsing.parseDefinitionAndResolveBubbles(DefinitionParsing.java:267)
        at org.kframework.kompile.Kompile.parseDefinition(Kompile.java:390)
        at org.kframework.kompile.Kompile.run(Kompile.java:213)
        at org.kframework.kompile.KompileFrontEnd.run(KompileFrontEnd.java:91)
        at org.kframework.main.FrontEnd.main(FrontEnd.java:60)
        at org.kframework.main.Main.runApplication(Main.java:127)
        at org.kframework.main.Main.runApplication(Main.java:117)
        at org.kframework.main.Main.main(Main.java:58)
[Error] Internal: Uncaught exception thrown of type AssertionError
(AssertionError: null)
```

--- a/k-distribution/tests/regression-new/checks/localDuplicateList.k
+++ b/k-distribution/tests/regression-new/checks/localDuplicateList.k
@@ -1,0 +1,10 @@
+module LOCALDUPLICATELIST-SYNTAX
+endmodule
+
+module LOCALDUPLICATELIST
+  imports BOOL-SYNTAX
+  imports INT-SYNTAX
+
+  syntax Foo ::= List{Bool, ","}
+  syntax Foo ::= List{Int, ";"}
+endmodule

--- a/k-distribution/tests/regression-new/checks/localDuplicateList.k.out
+++ b/k-distribution/tests/regression-new/checks/localDuplicateList.k.out
@@ -1,0 +1,5 @@
+[Error] Compiler: Sort Foo previously declared as a user list at Source(localDuplicateList.k) and Location(9,18,9,32)
+	Source(localDuplicateList.k)
+	Location(8,18,8,33)
+	8 |	  syntax Foo ::= List{Bool, ","}
+	  .	                 ^~~~~~~~~~~~~~~

--- a/kore/src/main/scala/org/kframework/definition/outer.scala
+++ b/kore/src/main/scala/org/kframework/definition/outer.scala
@@ -481,10 +481,12 @@ case class Module(
   def checkUserLists(): Unit = localSentences.foreach {
     case p @ Production(_, _, srt, _, atts) =>
       if (atts.contains(Att.USER_LIST)) {
-        val prev = importedSentences.find(s =>
+        val prev = sentences.find(s =>
           s.isInstanceOf[Production]
             && s.asInstanceOf[Production].sort.equals(srt)
             && s.att.contains(Att.USER_LIST)
+            && !(s.source.equals(p.source)
+              && s.location.equals(p.location))
         )
         if (prev.isDefined)
           throw KEMException.compilerError(

